### PR TITLE
feat(flags): support early_exit in local evaluation

### PR DESCRIPTION
**Pull request by gustavohstrassburger**
## :bulb: Motivation and Context

PostHog added an "early exit" option to feature flags in the server-side (Rust) evaluation engine. When a flag's `filters.early_exit` is `true`, local condition evaluation must STOP and return a definitive `false`/disabled as soon as a condition group's property filters match (or the group has no property filters) but the rollout percentage EXCLUDES the user — instead of falling through to evaluate later condition groups.

This was already ported to `posthog-node` and `posthog-python`; this PR ports the same behavior to `posthog-php` so local evaluation matches the server.

### Behavior

For each condition group, evaluation now distinguishes three outcomes:

- **MATCH** — property filters matched AND rollout included the user → flag matches (existing behavior).
- **NO_MATCH** — a property filter did NOT match → continue to the next group. Always falls through, even when `early_exit` is enabled.
- **OUT_OF_ROLLOUT_BOUND** — property filters matched (or none) but rollout EXCLUDED the user.

When `early_exit` is enabled AND a group's outcome is `OUT_OF_ROLLOUT_BOUND`, evaluation returns `false` immediately. When `early_exit` is unset or `false`, existing behavior is preserved exactly.

Implementation: `isConditionMatch` now returns a tri-state `ConditionMatch` enum (`Match` / `NoMatch` / `OutOfRolloutBound`) instead of a bool, letting `matchFeatureFlagProperties` short-circuit on rollout exclusion only.

## :green_heart: How did you test it?

Added unit tests to `FeatureFlagLocalEvaluationTest` covering: `early_exit` enabled returns `false` without evaluating a later matching group; `early_exit` unset falls through (regression); `early_exit` explicitly `false` falls through; and property-filter failure (not rollout) does NOT early-exit even when enabled.

Full suite green: `vendor/bin/phpunit` → 334 tests, 3362 assertions, 0 failures. `vendor/bin/phpcs` (PSR12) → 0 errors on changed files.

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [ ] I updated the docs if needed.
- [x] No breaking change or entry added to the changelog.

### If releasing new changes

- [x] Ran `pnpm changeset` to generate a changeset file

---
*Created with [PostHog Code](https://posthog.com/code?ref=pr)*

### CODE REVIEW

File: lib/FeatureFlag.php [hunk 1]
```diff
@@ -698,6 +701,11 @@ public static function matchFeatureFlagProperties(
                     } else {
                         return FeatureFlag::getMatchingVariant($flag, $effectiveBucketing) ?? true;
                     }
+                } elseif ($earlyExitEnabled && $matchResult === ConditionMatch::OutOfRolloutBound) {
```
Comment (dustinbyrne) [inline lib/FeatureFlag.php:704]: Blocking correctness issue: this returns a definitive `false` even if a previous condition was inconclusive.

`$isInconclusive` can already be true from an earlier condition, but this branch returns before the final inconclusive fallback runs. That means local evaluation can incorrectly disable the flag instead of falling back to server evaluation.

Repro shape:
1. earlier condition throws/catches `InconclusiveMatchException`
2. later condition is `OutOfRolloutBound`
3. `filters.early_exit=true`

Current result: `false`
Expected: throw `InconclusiveMatchException` / server fallback

Please guard this branch with `$isInconclusive` and throw/propagate inconclusive rather than returning `false`.

**Review by dustinbyrne — COMMENTED** [review]
Validated the early_exit edge case locally; leaving one blocking correctness note inline.

**Review by dustinbyrne — COMMENTED** [review]
Take a look at the following test, it's still a failing condition. Once the early exit / out of rollout band condition is reached, later groups shouldn't be considered. But because an earlier condition was inconclusive, local evaluation also can’t safely return false, so it should preserve the inconclusive/server-fallback path.

```php
public function testEarlyExitAfterInconclusiveDoesNotContinueToMatch(): void
{
    $flag = [
        "key" => "early-exit-flag",
        "filters" => [
            "early_exit" => true,
            "groups" => [
                [
                    "properties" => [["key" => "missing_prop", "value" => "x", "operator" => "exact"]],
                    "rollout_percentage" => 100,
                ],
                [
                    "properties" => [["key" => "region", "value" => "us", "operator" => "exact"]],
                    "rollout_percentage" => 0,
                ],
                [
                    "properties" => [["key" => "region", "value" => "us", "operator" => "exact"]],
                    "rollout_percentage" => 100,
                ],
            ],
        ],
    ];

    $this->expectException(InconclusiveMatchException::class);

    FeatureFlag::matchFeatureFlagProperties($flag, "test-user", ["region" => "us"]);
}
```

**Review by dustinbyrne — APPROVED** [review]
